### DISCUSSION TITLE
Different testCaseId when Retrying; Results in Duplicated Test Case

### DISCUSSION THREAD

**Discussion by Staicul** [allure-framework/allure2#2653] [Questions & Support]
We have a WebdriverIO TS test automation project. For mobile native tests (executed on SauceLabs), we added 2 retries. So, in case a test fails, it's retried (max 2 times) until it passes. 
The retry mechanism works, but reporting is broken. Every retry is considered by Allure a new test, and if a (Jenkins) job has tests that didn't pass on the first try, but passed on retries, the build is not green. 

The reporting works well for web. The test is marked as passed and retries appear in the retry tab. However, when running on real devices on SauceLabs, the retry is considered a new test case. Check attached screenshots. 
I was able to isolate the issue by going through the Allure result.json files and the problem seems to be that the retry gets a different 'testCaseId'. On web, the retry has the same 'testCaseId' as the original run.

Any explanations for this behaviour?
Is there a way to force the `testCaseId` to be the same for test cases with the same name? Or any other workaround?

<img width="2089" alt="Screenshot 2024-07-22 at 17 20 25" src="https://github.com/user-attachments/assets/aad76a7d-a299-4197-be47-061322eabe07">
<img width="2159" alt="Screenshot 2024-07-22 at 17 25 22" src="https://github.com/user-attachments/assets/82f01a0b-67ce-4363-9e19-6cf0a1a63b66">
<img width="2587" alt="Screenshot 2024-07-23 at 19 03 57" src="https://github.com/user-attachments/assets/db712de6-a834-4fc3-ae22-f0c0cbd94ae8">

**Comment by baev** [comment]
[From the code](https://github.com/webdriverio/webdriverio/blob/v8.39.1/packages/wdio-allure-reporter/src/utils.ts#L262), the `testCaseId` generation is wrong.

**Reply by baev** [reply]
https://github.com/webdriverio/webdriverio/pull/11532

**Reply by Staicul** [reply]
Thanks!

**Reply by baev** [reply]
https://github.com/webdriverio/webdriverio/issues/13215
